controllers/mirroring/mirroring_controller.go

```diff
@@ -531,7 +531,7 @@ func (r *MirroringReconciler) reconcileRadosNamespaceMirroring(
 			}
 			consumerOwner := &rns.OwnerReferences[consumerIndex]
 			consumer := storageConsumerByName[consumerOwner.Name]
-			if consumer == nil || consumer.Status.Client.ID == "" {
+			if consumer == nil || consumer.Status.Client == nil || consumer.Status.Client.ID == "" {
```

Comment:
What would be the case where we have the client struct but we don't have an id?

Reply:
I agree that it is not possible, removing it